Integration test flaky test - Authd tier 0-1

We observed a new error in authd integration tests tier 0-1. [first execution](https://github.com/wazuh/wazuh/actions/runs/25371816142/attempts/1), [second execution](https://github.com/wazuh/wazuh/actions/runs/25371816142/job/74413799489) in this issue [#35872](https://github.com/wazuh/wazuh/issues/35872)

These errors are related to using a non-defined attribute in `X509` object.

```text

_______ ERROR at setup of test_authd_ssl_certs[NO CERT - Verify Host NO] _______
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'
_____ ERROR at setup of test_authd_ssl_certs[VALID CERT - Verify Host NO] ______
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'
___ ERROR at setup of test_authd_ssl_certs[INCORRECT CERT - Verify Host NO] ____
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'
___ ERROR at setup of test_authd_ssl_certs[INCORRECT HOST - Verify Host NO] ____
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'
______ ERROR at setup of test_authd_ssl_certs[NO CERT - Verify Host YES] _______
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'
_____ ERROR at setup of test_authd_ssl_certs[VALID CERT - Verify Host YES] _____
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'
___ ERROR at setup of test_authd_ssl_certs[INCORRECT CERT - Verify Host YES] ___
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'
___ ERROR at setup of test_authd_ssl_certs[INCORRECT HOST - Verify Host YES] ___
test_authd/test_ssl/conftest.py:22: in generate_ca_certificate
    controller = CertificateController()
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:30: in __init__
    self.root_ca_cert = self._create_ca_cert(self.root_ca_key)
/usr/local/lib/python3.12/dist-packages/wazuh_testing/tools/certificate_controller.py:88: in _create_ca_cert
    cert.add_extensions([
E   AttributeError: 'X509' object has no attribute 'add_extensions'

```

Another error could be found while doing this issue [#36043](https://github.com/wazuh/wazuh/issues/36043): https://github.com/wazuh/wazuh/actions/runs/25727655254.

The error consists in a test timeout not stopping the python script workflow.

# DOD

- [ ] Propose a fix if apply

# 14/05/2026

Today I've been researching a bit on what is causing the second issue and ran some tests on my local machine

<details>
<summary> First test execution </summary>

```
=============================================== ERRORS ===============================================
_______________________ ERROR at setup of test_ossec_auth_messages[AgentName] ________________________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
______________________ ERROR at setup of test_ossec_auth_messages[Single Group] ______________________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
______________________ ERROR at setup of test_ossec_auth_messages[Multi Group] _______________________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
______________________ ERROR at setup of test_ossec_auth_messages[Specific IP] _______________________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
__________________ ERROR at setup of test_ossec_auth_messages[Error Duplicate name] __________________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
__________________ ERROR at setup of test_ossec_auth_messages[Error Invalid group] ___________________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
___________________ ERROR at setup of test_ossec_auth_messages[Error Duplicate IP] ___________________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
_____________ ERROR at setup of test_ossec_auth_messages[Agent message with key hash 1] ______________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
_____________ ERROR at setup of test_ossec_auth_messages[Agent message with key hash 2] ______________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
_________ ERROR at setup of test_authd_key_request_worker[Forward key request to master ID] __________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
_________ ERROR at setup of test_authd_key_request_worker[Forward key request to master IP] __________
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:510: in configure_sockets_environment_implementation
    mitm.start()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:325: in start
    self.run()
/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/mitm.py:308: in run
    os.rename(self.listener_socket_address, self.forwarded_socket_path)
E   FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock' -> '/var/ossec/queue/cluster/c-internal.sock.original'
_____ ERROR at setup of test_authd_use_password[Use password no - Register with wrong password] ______
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:502: in configure_sockets_environment_implementation
    services.wait_expected_daemon_status(running_condition=False)
/usr/local/lib/python3.10/dist-packages/wazuh_testing/utils/services.py:235: in wait_expected_daemon_status
    raise TimeoutError(f"{target_daemon} does not meet condition: running = {running_condition}")
E   TimeoutError: None does not meet condition: running = False
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
_ ERROR at setup of test_authd_use_source_ip[Use source ip yes, Let manager decide, use_source_ip enabled] _
conftest.py:573: in configure_sockets_environment
    yield from configure_sockets_environment_implementation(request)
conftest.py:502: in configure_sockets_environment_implementation
    services.wait_expected_daemon_status(running_condition=False)
/usr/local/lib/python3.10/dist-packages/wazuh_testing/utils/services.py:235: in wait_expected_daemon_status
    raise TimeoutError(f"{target_daemon} does not meet condition: running = {running_condition}")
E   TimeoutError: None does not meet condition: running = False
----------------------------------------- Captured log setup -----------------------------------------
DEBUG    wazuh_testing:conftest.py:349 Wazuh control set to True
DEBUG    wazuh_testing:conftest.py:353 Ignore error set to True
DEBUG    wazuh_testing:conftest.py:361 Restarting wazuh using wazuh-control
============================================== FAILURES ==============================================
___________________ test_agentd_server_configuration[Basic name and IPV6 short 17] ___________________
test_authd/test_api_registration/test_api_agent_registration.py:194: in test_agentd_server_configuration
    assert retrieve_client_key_entry(expected_client_keys_entry),\
E   AssertionError: Client keys expected {'name': 'api-registration-agent-19', 'ip': '0000:0000:0000:0000:0000:0000:0001:0008'} but no agent was found for that configuration
E   assert []
E    +  where [] = retrieve_client_key_entry({'ip': '0000:0000:0000:0000:0000:0000:0001:0008', 'name': 'api-registration-agent-19'})
========================================== warnings summary ==========================================
test_authd/test_common/test_authd.py: 32 warnings
test_authd/test_common/test_authd_agents_ctx.py: 6 warnings
test_authd/test_common/test_authd_key_hash.py: 5 warnings
test_authd/test_common/test_authd_valid_name_ip.py: 28 warnings
test_authd/test_force_options/test_authd_force_options.py: 49 warnings
test_authd/test_remote_enrollment/test_remote_enrollment.py: 9 warnings
test_authd/test_ssl/test_authd_ssl_certs.py: 8 warnings
test_authd/test_ssl/test_authd_ssl_options.py: 7 warnings
test_authd/test_use_password/test_authd_use_password.py: 22 warnings
test_authd/test_use_source_ip/test_authd_use_source_ip.py: 18 warnings
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/socket_controller.py:76: DeprecationWarning: ssl.PROTOCOL_TLSv1_2 is deprecated
    context = ssl.SSLContext(ssl_version)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
====================================== short test summary info =======================================
FAILED test_authd/test_api_registration/test_api_agent_registration.py::test_agentd_server_configuration[Basic name and IPV6 short 17] - AssertionError: Client keys expected {'name': 'api-registration-agent-19', 'ip': '0000:0000:0000:...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[AgentName] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Single Group] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Multi Group] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Specific IP] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Error Duplicate name] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Error Invalid group] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Error Duplicate IP] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Agent message with key hash 1] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_cluster/test_authd_worker.py::test_ossec_auth_messages[Agent message with key hash 2] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_key_request/test_authd_key_request_worker.py::test_authd_key_request_worker[Forward key request to master ID] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_key_request/test_authd_key_request_worker.py::test_authd_key_request_worker[Forward key request to master IP] - FileNotFoundError: [Errno 2] No such file or directory: '/var/ossec/queue/cluster/c-internal.sock...
ERROR test_authd/test_use_password/test_authd_use_password.py::test_authd_use_password[Use password no - Register with wrong password] - TimeoutError: None does not meet condition: running = False
ERROR test_authd/test_use_source_ip/test_authd_use_source_ip.py::test_authd_use_source_ip[Use source ip yes, Let manager decide, use_source_ip enabled] - TimeoutError: None does not meet condition: running = False
===== 1 failed, 150 passed, 2 skipped, 1 xfailed, 184 warnings, 13 errors in 10733.31s (2:58:53) =====
```

</details>

# 18/05/2026

Today I've applied some changes, executed tests in local and it seemed that everything was fine, but after the workflow execution it did not change anything.

# 19/05/2026

Today I've applied some more changes and apprently fixed the workflow not stopping its execution, but the flaky is still present.

# 20/05/2026

Today I've again applied some other changes and I have to still check if the flaky is present or not, first workflow execution has gone nice but some other executions must be done in order to see if the flaky is present or not.

# 21/05/2026

Today I applied more changes in order to try and fix the flaky, as it is still present in the workflow.

# 22/05/2026 

Today I applied another changed, executed the workflow 3 times and the 3 were succesful.

--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -495,53 +495,121 @@ def configure_sockets_environment_implementation(
     Args:
         request (pytest.FixtureRequest): Provide information about the current test function which made the request.
     """
+
     monitored_sockets_params = getattr(request.module, "monitored_sockets_params")
 
-    # Stop wazuh-service and ensure all daemons are stopped
-    services.control_service("stop")
-    services.wait_expected_daemon_status(running_condition=False)
-
     monitored_sockets = list()
-    mitm_list = list()
+    started_mitms = list()
+    started_daemons = list()
 
-    # Start selected daemons and monitored sockets MITM
-    for daemon, mitm, daemon_first in monitored_sockets_params:
-        not daemon_first and mitm is not None and mitm.start()
-        services.control_service("start", daemon=daemon, debug_mode=True)
-        services.wait_expected_daemon_status(
-            running_condition=True,
-            target_daemon=daemon,
-            extra_sockets=[mitm.listener_socket_address]
-            if mitm is not None and mitm.family == "AF_UNIX"
-            else [],
-        )
-        daemon_first and mitm is not None and mitm.start()
-        if mitm is not None:
-            monitored_sockets.append(
-                queue_monitor.QueueMonitor(monitored_object=mitm.queue)
+    try:
+        # Stop wazuh-service and ensure all daemons are stopped
+        try:
+            services.control_service("stop")
+        except Exception as e:
+            logger.warning(f"Setup step failed: {e}")
+
+        try:
+            services.wait_expected_daemon_status(running_condition=False)
+        except Exception as e:
+            logger.warning(f"Setup step failed: {e}")
+
+
+        # Clean leftover daemons. Missing files are the expected case when the
+        # previous teardown ran cleanly, so swallow FileNotFoundError silently.
+        for daemon, mitm, daemon_first in monitored_sockets_params:
+            pid_file = f"/var/ossec/var/run/{daemon}.pid"
+            try:
+                os.unlink(pid_file)
+            except FileNotFoundError:
+                pass
+
+            if mitm is not None and mitm.family == "AF_UNIX":
+                try:
+                    os.unlink(mitm.listener_socket_address)
+                except FileNotFoundError:
+                    pass
+
+                try:
+                    os.unlink(f"{mitm.listener_socket_address}.original")
+                except FileNotFoundError:
+                    pass
+
+                mitm.event.clear()
+
+        # Start selected daemons and monitored sockets MITM
+        for daemon, mitm, daemon_first in monitored_sockets_params:
+            if not daemon_first and mitm is not None:
+                mitm.start()
+                started_mitms.append(mitm)
+
+            services.control_service("start", daemon=daemon, debug_mode=True)
+            started_daemons.append(daemon)
+
+            # Use a 60s timeout (vs the framework default of 10s) because
+            # test_authd_key_request_worker has been observed to need >30s for
+            # wazuh-authd to publish its pid file when started right after the
+            # previous module killed wazuh-authd (likely TIME_WAIT on port 1515
+            # or post-fork init taking longer than expected). If the timeout
+            # still hits, the next step is to capture /var/ossec/logs/ossec.log
+            # to see what wazuh-authd is doing after goDaemon().
+            services.wait_expected_daemon_status(
+                target_daemon=daemon,
+                running_condition=True,
+                timeout=60,
+                extra_sockets=[mitm.listener_socket_address]
+                if mitm is not None and mitm.family == "AF_UNIX"
+                else [],
             )
-            mitm_list.append(mitm)
 
-    setattr(request.module, "monitored_sockets", monitored_sockets)
+            if daemon_first and mitm is not None:
+                mitm.start()
+                started_mitms.append(mitm)
 
-    yield
+            if mitm is not None:
+                monitored_sockets.append(
+                    queue_monitor.QueueMonitor(monitored_object=mitm.queue)
+                )
 
-    # Stop daemons and monitored sockets MITM
-    for daemon, mitm, _ in monitored_sockets_params:
-        mitm is not None and mitm.shutdown()
-        services.control_service("stop", daemon=daemon)
-        services.wait_expected_daemon_status(
-            running_condition=False,
-            target_daemon=daemon,
-            extra_sockets=[mitm.listener_socket_address]
-            if mitm is not None and mitm.family == "AF_UNIX"
-            else [],
-        )
+        setattr(request.module, "monitored_sockets", monitored_sockets)
 
-    # Delete all db
-    database.delete_dbs()
+        yield
+    finally:
+        # Stop daemons and monitored sockets MITM
+        for daemon, mitm, _ in monitored_sockets_params:
+            if mitm is not None and mitm in started_mitms:
+                try:
+                    mitm.shutdown()
+                except Exception as e:
+                    logger.warning(f"Cleanup step failed: {e}")
 
-    services.control_service("start")
+            if daemon in started_daemons:
+                try:
+                    services.control_service("stop", daemon=daemon)
+                except Exception as e:
+                    logger.warning(f"Cleanup step failed: {e}")
+
+                try:
+                    services.wait_expected_daemon_status(
+                        running_condition=False,
+                        target_daemon=daemon,
+                        extra_sockets=[mitm.listener_socket_address]
+                        if mitm is not None and mitm.family == "AF_UNIX"
+                        else [],
+                    )
+                except Exception as e:
+                    logger.warning(f"Cleanup step failed: {e}")
+
+        # Delete all db
+        try:
+            database.delete_dbs()
+        except Exception as e:
+            logger.warning(f"Cleanup step failed: {e}")
+
+        try:
+            services.control_service("start")
+        except Exception as e:
+            logger.warning(f"Cleanup step failed: {e}")
 
 
 @pytest.fixture(scope="module")

